paccmann_chemistry/models/vae.py

```diff
@@ -119,19 +122,51 @@ def encoder_train_step(self, input_seq):
 
         # Backward pass:
         if self.bidirectional:
-            assert len(input_seq.shape) == 2, 'Input Seq must be 2D Tensor.'
+            # assert len(input_seq.shape) == 2, 'Input Seq must be 2D Tensor.'
             hidden_backward = self.backward_stackgru.init_hidden()
             stack_backward = self.backward_stackgru.init_stack()
 
+            if self._packed_inputs:
+
+                input_seq = self.unpack_sequence(input_seq)
+
+                # REVIEW Idk if this is fast or slow, but I think that it's
+                # alright to go to list and reverse it there. Also, right now
+                # were okay with breaking hte graph, as this should be the
+                # first step on the entire graph, however it may break the
+                # entire thing in other situations (e.g. if the ebedding is
+                # done before hand)
+                input_seq = [torch.tensor(x.tolist()[::-1]) for x in input_seq]
```

Comment:
@jannisborn does this **REVIEW** note make sense?

Reply:
@niladell @jannisborn this might be slow because `tolist` will trigger a transfer GPU to CPU.
Maybe cleaning and handle this vith a view or indexing will not force a transfer but only do a copy in the GPU.

Reply:
Okay, that sounds reasonable. I think that @jannisborn was doing something along the lines of:
```python
idx = [i for i in range(input)seq.size(0)-1, -1, -1)]
idx = torch.LongTensor(idx)
inverted_tensor = tensor.index_select(0, idx)
```
so I'll do that. That way I won't break the graph (for whatever that may be useful in the future), right?

Reply:
The cleaning was a cloning :).
That said, this solution should work.

Reply:
Yes, I think that's better. It's a pity that torch does not suppport `[::-1]`. I think it's good practice to use torch operations whenever possible (aka only using numpy/lists etc when *really* necessary).

Reply:
> The cleaning was a cloning :)
Not sure I follow here.
> It's a pity that torch does not suppport `[::-1]`
Yeah, it is...

Reply:
Fixed in 3a29ef843128123ec6aa1601f82f401ba135df41